server/src/routes/users.router.ts

```diff
@@ -1,25 +1,29 @@
 import express from "express";
+import { ObjectId } from "mongodb";
 import { verifyToken } from "../utils/jwt";
 import { db } from "../mongodb";
 import { User } from "./register.router";
 
 const router = express.Router();
 
 // GET /users/info
-router.use("/info", verifyToken, async (req, res) => {
-    if (!req.userId) return res.status(401).json({ error: "Unauthorized" });
+router.get("/info", verifyToken, async (req, res) => {
     try {
-        const userInfo = await db
-            .collection<User>("users")
-            .findOne(
-                { _id: req.userId },
-                { projection: { password: 0, googleId: 0, githubId: 0 } }
-            );
-        if (!userInfo) return res.status(404).json({ error: "User not found" });
-        res.status(200).json(userInfo);
+        const userObjectId = typeof req.userId === "string" 
+            ? new ObjectId(req.userId) 
+            : req.userId;
+
+        const user = await db.collection<User>("users").findOne(
+            { _id: userObjectId },
+            { projection: { password: 0, googleId: 0, githubId: 0 } }
+        );
+        if (!user) {
+            return res.status(404).send({ message: "User not found" });
+        }
+        res.send(user);
     } catch (error) {
         console.error("Error fetching user info:", error);
-        res.status(500).json({ error: "Internal server error" });
+        res.status(500).send({ message: "Internal server error" });
```

Comment:
All endpoints have `res.status(500).json({ error: "Internal server error" });`, please revert this
```suggestion
        res.status(500).json({ error: "Internal server error" });
```